### PR TITLE
Implement NewSubdirectoryClient on *directory.Client

### PR DESCRIPTION
**Pull request by adreed-msft**
- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azdatalake/directory/client.go [hunk 1]
```diff
@@ -227,6 +227,28 @@ func (d *Client) NewFileClient(fileName string) (*file.Client, error) {
 	return (*file.Client)(base.NewPathClient(fileURL, newBlobURL, newBlobClient, d.generatedDirClientWithDFS().InternalClient().WithClientName(shared.FileClient), d.sharedKey(), d.identityCredential(), d.getClientOptions())), nil
 }
 
+func (d *Client) NewSubdirectoryClient(subDirectoryName string) (*Client, error) {
```
Comment (gapra-msft) [inline sdk/storage/azdatalake/directory/client.go:230]: Let's add some tests that get a subdirectory client - check the URL, do some API calls.

Comment (gapra-msft) [inline sdk/storage/azdatalake/directory/client.go:230]:
```suggestion
func (d *Client) NewSubdirectoryClient(subdirectoryName string) (*Client, error) {
```

Comment (adreed-msft) [reply sdk/storage/azdatalake/directory/client.go:230]: Accidentally pushed without the test I wrote!

File: sdk/storage/azdatalake/directory/client_test.go [hunk 2]
```diff
@@ -61,6 +62,30 @@ type UnrecordedTestSuite struct {
...
+	fsClient := svcClient.NewFileSystemClient(fsName)
+
+	dirName := testcommon.GenerateDirName(testName)
+	dirClient := fsClient.NewDirectoryClient(dirName)
+
+	subdirName := "subdir"
+	subdirClient, err := dirClient.NewSubdirectoryClient(subdirName)
+	_require.NoError(err)
+
+	correctDirURL := fmt.Sprintf("https://%s.dfs.core.windows.net/%s/%s/%s/", accountName, fsName, dirName, subdirName)
```
Comment (gapra-msft) [inline sdk/storage/azdatalake/directory/client_test.go:85]: Let's add like a create call and maybe a couple other calls and make this a Recorded test

**Comment by tanyasethi-msft** [comment]
Adding changes done in this PR to https://github.com/Azure/azure-sdk-for-go/pull/22176 to be a part of STG78-82 release of AzDatalake

**Pull request closed without merging**